apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetService.java

```diff
@@ -174,10 +177,27 @@ public Dataset findById(@NonNull UUID id) {
 
     @Override
     public Dataset findById(@NonNull UUID id, @NonNull String workspaceId) {
+        log.info("Finding dataset with id '{}', workspaceId '{}'", id, workspaceId);
         return template.inTransaction(READ_ONLY, handle -> {
             var dao = handle.attach(DatasetDAO.class);
+            var dataset =  dao.findById(id, workspaceId).orElseThrow(this::newNotFoundException);
+            log.info("Found dataset with id '{}', workspaceId '{}'", id, workspaceId);
+            return dataset;
+        });
+    }
 
-            return dao.findById(id, workspaceId).orElseThrow(this::createNotFoundError);
+    @Override
+    public List<Dataset> findByIds(@NonNull Set<UUID> ids, @NonNull String workspaceId) {
```

Comment:
Should we return only the name and ID? This probably would reduce the "overhead"

Reply:
It's an option, but I'd rather have and generic and re-usable method over something where we don't know the performance impact yet and what would be the improvement by following this approach.

We can always change this in the future.